src/python/ensembl/io/genomio/gff3/overlaps.py

```diff
@@ -0,0 +1,216 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Scan a GFF3 file to detect overlapping SeqFeature objects. Default object level => gene."""
+
+__all__ = [
+    "summarize_feature_stats",
+    "identify_feature_overlaps",
+    "scan_tree",
+    "get_intervals",
+]
+
+from collections import defaultdict
+import json
+import logging
+from pathlib import Path
+from pprint import pprint
+
+from BCBio import GFF
+from BCBio.GFF import GFFExaminer
+from Bio.SeqRecord import SeqRecord
+from intervaltree import Interval, IntervalTree
+
+from ensembl.io.genomio.utils import print_json
+from ensembl.utils.argparse import ArgumentParser
+from ensembl.utils.logging import init_logging_with_args
+
+
+def summarize_feature_stats(gff_in: Path) -> None:
+    """Analyse a GFF3 file and produce a summary of its feature types.
+
+    Args:
+        gff_in: User supplied GFF3 input file.
+    """
+
+    logging.info("Alt processing: Not parsing the GFF, producing summary feature stats instead!")
+
+    examiner = GFFExaminer()
+    with open(gff_in, "r", encoding="utf-8") as input_handle:
+        pprint(examiner.available_limits(input_handle))
+    input_handle.close()
+
+
+def identify_feature_overlaps(gff_in: Path, output_file: Path, isolate_feature: str) -> None:
+    """Detect overlapping GFF3 SeqFeature objects and dump to a report.
+
+    Args:
+        gff_in: User supplied GFF3 input file.
+        output_file: Output file to write feature overlaps.
+        isolate_feature: Sequence feature type to filter by.
+    """
+    logging.info("Processing sequence feature overlaps!")
+    logging.info(f"Output file = {str(output_file)}")
+    logging.info(f"Features filtered by type: {isolate_feature}")
+
+    gff_type_filter: dict = {"gff_type": [isolate_feature]}
+    seq_dict: dict = defaultdict(dict)
+    genes_dict: dict = {}
+    with open(gff_in, "r", encoding="utf-8") as input_handle:
```

Comment:
Same as before:
```suggestion
    with gff_in.open("r", encoding="utf-8") as input_handle:
```

Reply:
Does this one matter ? Asking for a friend ;P

Reply:
In a way it does not, in a way it does: just moving the code to the new(-ish) `pathlib` world.